rust/opendp/src/trans/mod.rs

```diff
@@ -104,55 +105,54 @@ pub struct BoundedSum<MI, MO, T> {
     data: PhantomData<T>,
 }
 
-// TODO: this is kind of ugly and should bubble results
-fn max<T: PartialOrd>(a: T, b: T) -> T {
-    match a.partial_cmp(&b) {
-        Some(Ordering::Less) => b,
-        _ => a
-    }
+fn max<T: PartialOrd>(a: T, b: T) -> Option<T> {
+    a.partial_cmp(&b).map(|o| if let Ordering::Less = o {b} else {a})
 }
 
-impl<MO, T> MakeTransformation2<VectorDomain<IntervalDomain<T>>, AllDomain<T>, HammingDistance, MO, T, T> for BoundedSum<HammingDistance, MO, T>
-    where T: 'static + Clone + PartialOrd + Sub<Output=T> + Mul<Output=T> + Div<Output=T> + Sum<T> + DistanceCast,
-          HammingDistance: Metric<Distance=u32>,
-          MO: SensitivityMetric<Distance=T> {
-    fn make2(lower: T, upper: T) -> Fallible<Transformation<VectorDomain<IntervalDomain<T>>, AllDomain<T>, HammingDistance, MO>> {
-        if lower > upper { return fallible!(MakeTransformation, "lower bound may not be greater than upper bound") }
+pub trait BoundedSumStability<MI: Metric, MO: Metric, T> {
+    fn get_stability(lower: MO::Distance, upper: MO::Distance) -> Fallible<StabilityRelation<MI, MO>>;
+}
```

Comment:
This was a cool trick, and the fancy trait bound made it possible to do our crazy dispatch!